### PR TITLE
added simple global meta viewer

### PR DESCRIPTION
**Pull request by jh87692**
### What does this PR do?

Simply displays the global object of the sigmf metadata below the spectrogram
It will truncate the description if > 50 characters and displays the full description on mouse over (description)

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you successfully ran tests with your changes locally? Including container validation.
* [x] Have you lint your code locally prior to submission?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: client/src/Components/MetaViewer/__tests__/MetaViewer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        "datatype": "datatype test",
+        "version": "version test",
+        "offset": 0,
+        "sample_rate": 0,
+        "description": "description test"
+    };
+
+    test('Renders correctly', async () => {
+        render(<MetaViewer {...meta} />);
+        expect(await screen.findByText('datatype test')).toBeInTheDocument();
```
Comment (Jaydee123) [inline client/src/Components/MetaViewer/__tests__/MetaViewer.test.tsx:20]: Can we add the other values to this?

File: client/src/Components/MetaViewer/MetaViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export const MetaViewer: React.FC<MetaViewerProps> = (meta: MetaViewerProps) => {
+    const truncateDescription = (desc: string) => {
+        return desc.length > 50 ? desc.substring(0, 50) + "..." : desc;
+    }
+    return ( 
+        <div className="border border-primary ml-3 mt-2">
+            <div className="flex mt-3 mb-3 stats shadow">
+            <div className="stat place-items-center">
```
Comment (Jaydee123) [inline client/src/Components/MetaViewer/MetaViewer.tsx:18]: Small thing, is the alignment of these divs right? Should it be indented one more?

File: client/src/Components/MetaViewer/__tests__/MetaViewer.test.tsx (hunk 1 above)
Comment (Jaydee123) [reply client/src/Components/MetaViewer/__tests__/MetaViewer.test.tsx:20]: As well as one for the truncate logic :D

Comment (jh87692) [reply client/src/Components/MetaViewer/__tests__/MetaViewer.test.tsx:20]: done

File: client/src/Components/MetaViewer/MetaViewer.tsx (hunk 2 above)
Comment (jh87692) [reply client/src/Components/MetaViewer/MetaViewer.tsx:18]: it seems to be for me

File: client/src/Components/MetaViewer/MetaViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+export interface MetaViewerProps {
+    "datatype": string;
+    "version": string;
+    "offset": number;
+    "sample_rate": number;
+    "description": string;
+}
+
+export const MetaViewer: React.FC<MetaViewerProps> = (meta: MetaViewerProps) => {
```
Comment (Nepomuceno) [inline client/src/Components/MetaViewer/MetaViewer.tsx:11]: Why not just receive the while meta here ?

File: client/src/Components/MetaViewer/__tests__/MetaViewer.test.tsx (hunk 1 above)
Comment (Nepomuceno) [reply client/src/Components/MetaViewer/__tests__/MetaViewer.test.tsx:20]: We already have a truncated logic for the metadata description I think in the sigmf metadata class

File: client/src/Components/Spectrogram/SpectrogramPage.tsx [hunk 4]
```diff
@@ -525,9 +526,15 @@ export const SpectrogramPage = () => {
                 )}
               </div>
             </div>
+            <MetaViewer
```
Comment (Nepomuceno) [inline client/src/Components/Spectrogram/SpectrogramPage.tsx:529]: I would just pass meta as the parameters and use the methods inside to grab the data they are a bit more reliable and take care or hard coded string

File: client/src/Components/MetaViewer/MetaViewer.tsx (hunk 3 above)
Comment (jh87692) [reply client/src/Components/MetaViewer/MetaViewer.tsx:11]: if you mean the whole meta? It's because only 5 things are needed from the object

Comment (Nepomuceno) [reply client/src/Components/MetaViewer/MetaViewer.tsx:11]: I know but passing the whole meta allow you to access the methods of the meta too we might pass only these properties but I iamgine if we look at the info pane right nwo it does look at other properties of the document.

File: client/src/Components/Spectrogram/SpectrogramPage.tsx (hunk 4 above)
Comment (jh87692) [reply client/src/Components/Spectrogram/SpectrogramPage.tsx]: yep, done

**Review by Nepomuceno — APPROVED** [review]
LGTM

**Review by Jaydee123 — APPROVED** [review]

**Pull request merged**